connect.py

```diff
@@ -17,15 +18,20 @@ def request(self, resource, host, itemtype, port=70):
                 response = socket_conn.makefile(mode = 'rb', errors = 'ignore')
             else:
                 response = socket_conn.makefile(mode = 'r', errors = 'ignore')
-        except:
+        except socket.timeout:
+            print('Socket timeout')
             socket_conn.close()
-            return False
+            return {'type': '3', 'body': '3ERROR: Server request timed out\tfalse\tnull.host\t1'}
+        except Exception as e:
+            print('Misc socket error: ', e)
+            socket_conn.close()
+            return {'type': '3', 'body': '3ERROR: Unable to communicate with remote server\tfalse\tnull.host\t1'}
 
         try:
             self.raw_response = response.read()
             self.filetype = itemtype
         except UnicodeDecodeError:
```

Comment:
I don't believe this is necessary anymore and is a relic from an earlier point in the code where I was handling a decode here (it has been moved to elsewhere in the code).